internal/secant/attest.go

```diff
@@ -58,14 +59,34 @@ func NewStatement(digest name.Digest, predicate io.Reader, ptype string) (*types
 
 // Attest is roughly equivalent to cosign attest.
 // The only real implementation of types.CosignerSignerVerifier is fulcio.SignerVerifier.
-func Attest(ctx context.Context, statements []*types.Statement, sv types.CosignerSignerVerifier, rekorClient *client.Rekor, ropt []remote.Option) error {
+func Attest(ctx context.Context, conflict string, statements []*types.Statement, sv types.CosignerSignerVerifier, rekorClient *client.Rekor, ropt []remote.Option) error {
 	digest := statements[0].Digest
 
 	// We don't actually need to access the remote entity to attach things to it
 	// so we use a placeholder here.
 	ropts := []ociremote.Option{ociremote.WithRemoteOptions(ropt...)}
 	se := ociremote.SignedUnknown(digest, ropts...)
 
+	atts, err := se.Attestations()
+	if err != nil {
+		return fmt.Errorf("fetching attestations: %w", err)
+	}
+
+	sigs, err := atts.Get()
+	if err != nil {
+		return fmt.Errorf("getting attestations: %w", err)
+	}
+
+	statements, err = newStatements(statements, sigs, conflict)
+	if err != nil {
+		return fmt.Errorf("failed to determine new statements: %w", err)
+	}
+
+	// If there are no net new statements, we can skip the write entirely.
+	if len(statements) == 0 {
```

Comment:
shouldn't this check `conflict == SKIPSAME`?

Reply:
oh you pass conflict into newStatements

Reply:
I don't think we need to. `newStatements` would only return a zero length result if you passed in a zero length statements, which... I think should be a no-op.